gobblin-api/src/main/java/gobblin/configuration/State.java

```diff
@@ -98,6 +102,19 @@ public void setProp(String key, Object value) {
   }
 
   /**
+   * Appends the input value to a list property that can be retrieved with {@link #getPropAsList}.
+   * @param key property key
+   * @param value property value (should not include commas, which are used as separators)
```

Comment:
value can include commas, if the user wants to pass a set of comma-separated values.
Also is it better to make the separator a parameter?